src/ralph/virtual/api.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+from rest_framework import serializers
+
+from ralph.api import RalphAPISerializer, RalphAPIViewSet, router
+from ralph.assets.api.serializers import BaseObjectSerializer
+from ralph.data_center.models.networks import IPAddress
+from ralph.data_center.models.physical import DataCenterAsset
+from ralph.virtual.models import (
+    CloudFlavor,
+    CloudHost,
+    CloudProject,
+    CloudProvider
+)
+
+
+class IPAddressSimpleSerializer(RalphAPISerializer):
+    class Meta:
+        model = IPAddress
+        depth = 1
+        fields = ['address', 'hostname']
+
+
+class CloudFlavorSimpleSerializer(RalphAPISerializer):
+    class Meta:
+        model = CloudFlavor
+        fields = ['name', 'tags', 'cores', 'memory', 'disk', 'url']
+
+
+class CloudHostSimpleSerializer(BaseObjectSerializer):
+    ip_addresses = serializers.ListField()
+
+    class Meta:
+        model = CloudHost
+        fields = ['hostname', 'ip_addresses', 'url']
+
+
+class CloudProjectSimpleSerializer(BaseObjectSerializer):
+    class Meta:
+        model = CloudProject
+        fields = ['name', 'tags', 'url']
+
+
+class DataCenterAssetSimpleSerializer(RalphAPISerializer):
+    class Meta:
+        model = DataCenterAsset
+        fields = ['hostname', 'tags', 'url']
+
+
+class SaveCloudFlavorSerializer(RalphAPISerializer):
+    cores = serializers.IntegerField()
+    memory = serializers.IntegerField()
+    disk = serializers.IntegerField()
+
+    def create(self, validated_data):
```

Comment:
It works with updates. The workaround for create is necessary cause otherwise method _set_component (CloudFlavor model) will try to assign components to object, which doesn't exist in DB.